dotnet/src/SemanticKernel.Abstractions/Reliability/HttpRetryConfig.cs

```diff
@@ -59,7 +59,11 @@ public int MaxRetryCount
         (HttpStatusCode)HttpStatusCodeType.RequestTimeout,
         (HttpStatusCode)HttpStatusCodeType.ServiceUnavailable,
         (HttpStatusCode)HttpStatusCodeType.GatewayTimeout,
-        (HttpStatusCode)HttpStatusCodeType.TooManyRequests
+        (HttpStatusCode)HttpStatusCodeType.TooManyRequests,
+        (HttpStatusCode)HttpStatusCodeType.InternalServerError,
+        (HttpStatusCode)HttpStatusCodeType.NotImplemented,
```

Comment:
Thoughts on NotImplemented and InsufficientStorage? Those feel less likely to be transient issues. Curious if you have some scenarios where these have been helpful.